### ISSUE TITLE
multicore-sys-monitor@ccadeptic23 ...problem loading the selected item, and it has been disabled.

### ISSUE THREAD

**Issue by RiaanBurger** [linuxmint/cinnamon-spices-applets#577]
@ccadeptic23
@jaszhix 

#515 (version 1.5.1) broke this applet for me with UI error "There was a problem loading the selected item, and it has been disabled." when one attempts to enable it (Mint 18.1 in a VM) or have just updated it (my Mint 17 desktop). 

The logs are extensive and include:

glass.log

```
error t=2017-05-02T19:57:44.247Z Failed to load prefs.json: GLib.FileError: Failed to open file '/home/riaan/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/prefs.json': No such file or directory Attempting to create it.
error t=2017-05-02T19:57:44.265Z Mainloop.idle_add_full is not a function
trace t=2017-05-02T19:57:44.265Z 
<----------------
MyApplet.prototype._init@/home/riaan/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/applet.js:93
MyApplet@/home/riaan/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/applet.js:56
main@/home/riaan/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/applet.js:327
createApplet@/usr/share/cinnamon/js/ui/appletManager.js:329
addAppletToPanels@/usr/share/cinnamon/js/ui/appletManager.js:256
finishExtensionLoad@/usr/share/cinnamon/js/ui/appletManager.js:59
loadExtension@/usr/share/cinnamon/js/ui/extension.js:451
reloadExtension@/usr/share/cinnamon/js/ui/extension.js:532
CinnamonDBus.prototype.ReloadXlet@/usr/share/cinnamon/js/ui/cinnamonDBus.js:293
_handleMethodCall@resource:///org/gnome/gjs/modules/overrides/Gio.js:261
_wrapJSObject/<@resource:///org/gnome/gjs/modules/overrides/Gio.js:331
---------------->
error t=2017-05-02T19:57:44.265Z [Applet "multicore-sys-monitor@ccadeptic23"]: Failed to evaluate 'main' function on applet: multicore-sys-monitor@ccadeptic23/15
error t=2017-05-02T19:57:44.266Z Could not load applet multicore-sys-monitor@ccadeptic23
```

**Comment by jaszhix** [comment]
What version of Cinnamon are you using?

**Comment by RiaanBurger** [comment]
On the bare metal desktop:

```
uname -a && dpkg -s cinnamon | grep Version
Linux riaan-desktop 4.4.0-59-generic #80~14.04.1-Ubuntu SMP Fri Jan 6 18:02:02 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Version: 2.8.8+rosa
```

In the VM:

```
uname -a && dpkg -s cinnamon | grep Version
Linux riaan-desktop 4.4.0-75-generic #96-Ubuntu SMP Thu Apr 20 09:56:33 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Version: 3.2.7+serena
```

I'll keep an eye on the issue, so please ask if there's anything I can help with.

**Comment by jaszhix** [comment]
I will have to convert it into a multi-version applet. This is a regression from my 3.4 compatibility fix that affects Cinnamon <= 3.0.

**Comment by RiaanBurger** [comment]
That was awesomely fast Jason, thank you very much. Works on both installs now.
